libraries/abstractions/common_io/include/iot_uart.h

```diff
@@ -108,74 +109,102 @@ typedef enum
+    uint8_t ucStopbits;    /**< The stop bits to be set for the UART port: 0 is no stop bits and 1 is stop bits. */
+    uint8_t ucFlowControl; /**< The flow control to be set for the UART port: 0 is disabled and 1 is enabled. */
 } IotUARTConfig_t;
 
 /**
- * @brief Initializes the UART peripheral of the board with default configuration.
+ * @brief Initializes the UART peripheral of the board.
  *
  * The application should call this function to initialize the desired UART port.
  *
+ * @warning Once opened, the same UART instance must be closed before calling open again.
+ *
  * @param[in] lUartInstance The instance of the UART port to initialize.
  *
- * @return handle to the UART if everything succeeds, otherwise NULL.
+ * @return
+ * - 'the handle to the UART port (not NULL)', on success.
+ * - 'NULL', if
+ *     - invalid instance number
+ *     - open same instance more than once before closing it
  */
 IotUARTHandle_t iot_uart_open( int32_t lUartInstance );
 
 /**
  * @brief Sets the application callback to be called on completion of an operation.
  *
- * On completion of receive or transmit on UART, the application is notified with a
- * function callback with the state of the transaction. The callback function and the parameters
- * for the state of the callback are set using iot_uart_set_callback.
- * The application should call this function with the callback function and its parameters.
+ * The callback is guaranteed to be invoked when the current asynchronous operation completes, either successful or failed.
+ * This simply provides a notification mechanism to user's application. It has no impact if the callback is not set.
+ *
+ * @note This callback will not be invoked when synchronous operation completes.
+ * @note This callback is per handle. Each instance has its own callback.
+ * @note Single callback is used for both read_async and write_async. Newly set callback overrides the one previously set.
+ * @warning If the input handle is invalid, this function silently takes no action.
  *
  * @param[in] pxUartPeripheral The peripheral handle returned in the open() call.
- * @param[in] xCallback The callback function to be called on completion of transaction.
- * @param[in] pvUserContext The user context to be passed back when callback is called.
+ * @param[in] xCallback The callback function to be called on completion of transaction (This can be NULL).
+ * @param[in] pvUserContext The user context to be passed back when callback is called (This can be NULL).
  */
 void iot_uart_set_callback( IotUARTHandle_t const pxUartPeripheral,
                             IotUARTCallback_t xCallback,
                             void * pvUserContext );
 
 /**
- * @brief Starts receiving the data from UART synchronously i.e. this is a
- * blocking call.
+ * @brief Starts receiving the data from UART synchronously.
+ *
+ * This function attempts to read certain number of bytes from slave device to a pre-allocated buffer, in synchronous way.
```

Comment:
There is a no slave device in UART. Can we please change this to "read certain number of bytes from receiver"?

Reply:
Oh yeah. I copied some words from I2C doc. I will remove all the slave. Thanks for catching this.

Reply:
And I believe you meant "read certain number of bytes from transmitter"